File: day05/day5.py
import sys
i=[int(c)for c in sys.argv[1].split("\n")]
l=i[:]
def b(a):
	s=0
	r=0
	while s<len(l):
		n=l[s]
		l[s]+=a(n)
		s+=n
		r+=1
	print(r)
b(lambda x:1)
l=i
b(lambda x:(1if x<3else-1))

File: day05/day5.swift
var a = CommandLine.arguments[1].split(separator: "\n").map {Int($0)!}
var i = 0, s = 0
while i<a.count {
    let n = a[i]
    a[i] += 1
    i += n
    s += 1
}
print(s)

a = CommandLine.arguments[1].split(separator: "\n").map {Int($0)!}
i = 0
s = 0
while i<a.count {
    let n = a[i]
    a[i] += (n<3 ? 1 : -1)
    i += n
    s += 1
}
print(s)
